### PR TITLE
(PC-12843) feat(accessibility): Hide accordion's children when not expanded

### PR DESCRIPTION
**Pull request by aliraiki**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-12843

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/ui/components/AccordionItem.tsx [hunk 1]
```diff
@@ -67,12 +68,16 @@ export const AccordionItem = ({
...
+    !open && setShowChildren(true) // Display children before opening animation begins
     Animated.timing(animatedController, {
       duration: 300,
       toValue: open ? 0 : 1,
       easing: Easing.bezier(0.4, 0.0, 0.2, 1),
       useNativeDriver: false,
-    }).start(() => setOpen((prevOpen) => !prevOpen))
+    }).start(() => {
+      setOpen((prevOpen) => !prevOpen)
+      open && setShowChildren(false) // Hide children after closing animation ends
```
Comment (kopax-polyconseil) [inline src/ui/components/AccordionItem.tsx:79]: j'aurai mis ça dans la function du dessus, quitte à rajouter des bracket, histoire d'être sur que `open` soit à jour.

File: src/ui/components/AccordionItem.tsx [hunk 2]
```diff
@@ -100,12 +105,13 @@ export const AccordionItem = ({
...
       </TouchableWithoutFeedback>
       <StyledAnimatedView style={{ height: bodyHeight }} testID="accordionBody">
-        <View
-          style={[styles.bodyContainer, bodyStyle]}
+        <StyledView
+          style={bodyStyle}
           testID="accordionBodyContainer"
-          onLayout={(event) => setBodySectionHeight(event.nativeEvent.layout.height)}>
+          onLayout={(event) => setBodySectionHeight(event.nativeEvent.layout.height)}
+          showChildren={showChildren}>
```
Comment (kopax-polyconseil) [inline src/ui/components/AccordionItem.tsx:112]: Je trouve le naming trop fonctionnelle pour définir un `display: none`, pourquoi pas un `hidden` ?

File: src/ui/components/AccordionItem.tsx [hunk 3]
```diff
@@ -120,16 +126,18 @@ const styles = StyleSheet.create({
...
+  display: showChildren ? 'flex' : 'none',
+  position: 'absolute',
+  bottom: 0,
+  width: '100%',
+  paddingBottom: getSpacing(6),
+  paddingHorizontal: getSpacing(6),
+  paddingTop: 0,
+}))
+
 const Title = styled(Typo.Title4).attrs(() => getTitleAttrs(2))({
```
Comment (kopax-polyconseil) [inline src/ui/components/AccordionItem.tsx:141]: J'ai raté un épisode, C'est quoi ce `2` ?

Comment (kopax-polyconseil) [reply src/ui/components/AccordionItem.tsx:143]: Je propose de renommer `getTitleAttrs` -> `getHeading(2)`/`getHeadingTitle(2)`/`getHeadingAttrs(2)`/`getHeadingTitleAttrs(2)`  , car on génère un heading 2, le title étant `this`

Comment (aliraiki) [reply src/ui/components/AccordionItem.tsx:143]: `getHeadingAttrs` me convient, je fais la modif

**Review by kopax-polyconseil — APPROVED** [review]

**Pull request merged**